freqtrade/exchange/binance.py

```diff
@@ -98,38 +102,72 @@ def additional_exchange_init(self) -> None:
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
 
-    async def _async_get_historic_ohlcv(
+    def get_historic_ohlcv(
         self,
         pair: str,
         timeframe: str,
         since_ms: int,
         candle_type: CandleType,
         is_new_pair: bool = False,
-        raise_: bool = False,
         until_ms: Optional[int] = None,
-    ) -> OHLCVResponse:
+    ) -> DataFrame:
         """
         Overwrite to introduce "fast new pair" functionality by detecting the pair's listing date
         Does not work for other exchanges, which don't return the earliest data when called with "0"
         :param candle_type: Any of the enum CandleType (must match trading mode!)
         """
         if is_new_pair:
-            x = await self._async_get_candle_history(pair, timeframe, candle_type, 0)
+            x = self.loop.run_until_complete(
+                self._async_get_candle_history(pair, timeframe, candle_type, 0)
+            )
             if x and x[3] and x[3][0] and x[3][0][0] > since_ms:
                 # Set starting date to first available candle.
                 since_ms = x[3][0][0]
                 logger.info(
                     f"Candle-data for {pair} available starting with "
                     f"{datetime.fromtimestamp(since_ms // 1000, tz=timezone.utc).isoformat()}."
                 )
+                if until_ms and since_ms >= until_ms:
+                    logger.warning(
+                        f"No available candle-data for {pair} before"
+                        f"{dt_from_ts(until_ms).isoformat()}"
+                    )
+                    return DataFrame(columns=DEFAULT_DATAFRAME_COLUMNS)
 
-        return await super()._async_get_historic_ohlcv(
+        if timeframe in ["1s", "1m", "5m"] and candle_type in [CandleType.SPOT, CandleType.FUTURES]:
+            df = self.loop.run_until_complete(
+                binance_public_data.fetch_ohlcv(
```

Comment:
i would rename the function (it's not fetching - it really is doing a whole download and a lot more) - and import it as function instead of the whole module ...

Reply:
From a caller's perspective, it just returns the ohlcv requested. Of course `fetch_ohlcv` can be confused with some method in CCXT so I imported the module name for readability.

How about `download_archive_ohlcv`(sounds not good ...)?  I'd like to hear your advice for the naming.

Reply:
`download_archive_ohlcv` sounds pretty good, actually - and explains pretty well what it does

Reply:
Renamed